### PR TITLE
Fixes 142: support cert via env variable

### PR DESCRIPTION
**Pull request by jlsherrill**
as well as via filename

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -105,6 +107,37 @@ func Load() {
...
+	cert, err := ConfigureCertificate()
+	if err != nil {
+		log.Fatal().Err(err).Msg("Could not read or parse cdn certificate.")
+	}
+	LoadedConfig.Certs.CdnCertPair = cert
+}
+
+const RhCertEnv = "RH-CDN-CERT-PAIR"
+
+func ConfigureCertificate() (*tls.Certificate, error) {
```
Comment (jlsherrill) [inline pkg/config/config.go:119]: Need to document this method

**Comment by jlsherrill** [comment]
TODO:  update deployment.yaml

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-142

**Comment by jlsherrill** [comment]
did deployment changes as part of a 2nd pr:  https://github.com/content-services/content-sources-backend/pull/71

**Comment by rverdile** [comment]
I'm not sure how to test this, could you add a couple of testing steps?

**Comment by jlsherrill** [comment]
oh yes, that would be helpful!

First import the repos:

```
make repos-import
```

I'll send you a keypair to use.  Today you can set cert_path in your config.yaml to point to the cert:
```
certs:
  cert_path: "/home/jlsherri/cdncert/cert.pemz"
```

so when you run 'go run cmd/external-repos/main.go introspect-all`  it will use that cert to fetch red hat repo metadata.

With this pr, that should still work, but if you specify the cert contents via an ENV variable, it will use that instead:

```
export RH_CDN_CERT_PAIR=`cat /home/jlsherri/cdncert/cert.pem`
go run cmd/external-repos/main.go introspect-all
```
and will ignore whatever cert_path is set to.  If you don't set the env variable, it should still use cert_path

**Review by rverdile — APPROVED** [review]
looks good

was able to use a cert via env variable and introspect all the repos. got the right error without the env variable.

**Comment by avisiedo** [comment]
Launched from inside the pod by exporting the env variable by hand before executing the command:

```raw
sh-4.4$ time ./external-repos introspect-all
{"level":"error","error":"Config File \"config\" Not Found in \"[/configs]\"","time":"2022-08-05T20:04:38Z"}
{"level":"debug","time":"2022-08-05T20:04:38Z","message":"Introspecting http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/"}
{"level":"debug","time":"2022-08-05T20:09:06Z","message":"Introspecting http://mirror.centos.org/centos/8-stream/BaseOS/x86_64/os/"}
{"level":"debug","time":"2022-08-05T20:10:36Z","message":"Introspecting http://mirror.centos.org/centos/8-stream/extras/x86_64/os/"}
{"level":"debug","time":"2022-08-05T20:10:36Z","message":"Introspecting http://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/"}
{"level":"debug","time":"2022-08-05T20:12:13Z","message":"Introspecting http://mirror.stream.centos.org/9-stream/BaseOS/x86_64/os/"}
{"level":"debug","time":"2022-08-05T20:12:28Z","message":"Introspecting https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/ansible/2/os"}
{"level":"debug","time":"2022-08-05T20:12:28Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel8/8.4/x86_64/appstream/os"}
{"level":"debug","time":"2022-08-05T20:16:33Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel8/8.4/x86_64/baseos/os"}
{"level":"debug","time":"2022-08-05T20:17:32Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel8/8.5/x86_64/appstream/os"}
{"level":"debug","time":"2022-08-05T20:22:49Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel8/8.5/x86_64/baseos/os"}
{"level":"debug","time":"2022-08-05T20:24:02Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel8/8.6/x86_64/appstream/os"}
{"level":"debug","time":"2022-08-05T20:30:53Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel8/8.6/x86_64/baseos/os"}
{"level":"debug","time":"2022-08-05T20:32:18Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel9/9.0/x86_64/appstream/os"}
{"level":"debug","time":"2022-08-05T20:32:36Z","message":"Introspecting https://cdn.redhat.com/content/dist/rhel9/9.0/x86_64/baseos/os"}
{"level":"debug","time":"2022-08-05T20:32:40Z","message":"Introspecting https://packages.cloud.google.com/yum/repos/cloud-sdk-el8-x86_64"}
{"level":"debug","time":"2022-08-05T20:32:48Z","message":"Introspecting https://packages.cloud.google.com/yum/repos/cloud-sdk-el9-x86_64"}
{"level":"debug","time":"2022-08-05T20:32:54Z","message":"Introspecting https://packages.cloud.google.com/yum/repos/google-compute-engine-el8-x86_64-stable"}
{"level":"debug","time":"2022-08-05T20:32:54Z","message":"Introspecting https://packages.cloud.google.com/yum/repos/google-compute-engine-el9-x86_64-stable"}
{"level":"debug","time":"2022-08-05T20:32:54Z","message":"Successfully Inserted 94614 packages"}

real	28m16.073s
user	7m47.465s
sys	0m33.434s

```

LGTM

**Review by swadeley — APPROVED** [review]
ACK

**Pull request merged**